meteor-app/imports/ui/ordering/cart-layout.js

```diff
@@ -9,29 +9,39 @@ Session.set('searchQuery', '')
 export default withTracker((props) => {
   const ordersHandle = Meteor.subscribe('all.orders')
   const removePart = async (OrderId, partId) => {
-    const orderedParts = await Orders.findOne({ _id: OrderId}).orderedParts
-    orderedParts.forEach(part => {
+    const order = await Orders.findOne({ _id: OrderId})
+    order.orderedParts.forEach(part => {
       if(part.partId === partId){
-        Meteor.callAsync('orders.removePart', OrderId, part)
+        order.totalPrice = order.totalPrice - (part.price * part.qty)
+        Meteor.callAsync('order.update', order._id, order)
```

Comment:
It doesn't make sense to call 2 Meteor methods in succession - create a new method and do both operations in the same method.
In fact, you are looping over these - you should be doing the whole thing in the server, not iterating over data and doing multiple server calls